src/colmap/estimators/cost_functions.h

```diff
@@ -56,26 +56,32 @@ inline Eigen::MatrixXd SqrtInformation(const Eigen::MatrixXd& covariance) {
 template <typename CameraModel>
 class ReprojErrorCostFunction {
  public:
-  explicit ReprojErrorCostFunction(const Eigen::Vector2d& point2D)
-      : observed_x_(point2D(0)), observed_y_(point2D(1)) {}
-
-  static ceres::CostFunction* Create(const Eigen::Vector2d& point2D) {
+  explicit ReprojErrorCostFunction(
+      const Eigen::Vector2d& point2D,
+      const Eigen::Matrix2d& point2D_covar = Eigen::Matrix2d::Identity())
+      : observed_x_(point2D(0)),
+        observed_y_(point2D(1)),
+        sqrt_information_point_(SqrtInformation(point2D_covar)) {}
```

Comment:
It would be much cleaner to have a wrapper cost function that can be bypassed if the covariance is identity, such that the performance is degraded only for non-default covariances. I looked into this a while ago and the [WeightedCostFunction](https://groups.google.com/g/ceres-solver/c/OobKeuuvXHE) appears to be only solution. This might be easier to implement now than in 2013... maybe worth reviving that thread? or use the `IsotropicNoiseCostFunctionWrapper` for now.